src/EnergyPlus/General.cc

```diff
@@ -288,42 +288,33 @@ void SolveRoot(const EnergyPlusData &state,
     XRes = XTemp;
 }
 
-void MovingAvg(EPVector<Real64> &DataIn, int const NumItemsInAvg)
-{
-    if (NumItemsInAvg <= 1) return; // no need to average/smooth
-
-    EPVector<Real64> TempData;
-    TempData.allocate(2 * DataIn.size()); // a scratch array twice the size, bottom end duplicate of top end
-
-    for (std::size_t i = 1; i <= DataIn.size(); ++i) {
-        TempData(i) = TempData(DataIn.size() + i) = DataIn(i); // initialize both bottom and top end
-        DataIn(i) = 0.0;
-    }
-
-    for (std::size_t i = 1; i <= DataIn.size(); ++i) {
-        for (int j = 1; j <= NumItemsInAvg; ++j) {
-            DataIn(i) += TempData(DataIn.size() - NumItemsInAvg + i + j); // sum top end including NumItemsInAvg history terms
-        }
-        DataIn(i) /= NumItemsInAvg; // average to smooth over NumItemsInAvg window
-    }
-}
-
 void MovingAvg(Array1D<Real64> &DataIn, int const NumItemsInAvg)
 {
     if (NumItemsInAvg <= 1) return; // no need to average/smooth
-
-    Array1D<Real64> TempData(2 * DataIn.size()); // a scratch array twice the size, bottom end duplicate of top end
-
-    for (std::size_t i = 1; i <= DataIn.size(); ++i) {
-        TempData(i) = TempData(DataIn.size() + i) = DataIn(i); // initialize both bottom and top end
-        DataIn(i) = 0.0;
+    //if (std::accumulate(DataIn.begin(), DataIn.end(), 0.0) == 0.0) {
+    //    return; // no noeed to average if all zeroes
+    //}
```

Comment:
Pushing up the new `MovingAvg`. The e-17'ish diffs are causing some big diffs in 5ZoneAirCooledWithDOASAirLoop locally (due to x/1e-17) in some system sizing calcs. Thought maybe checking for all zeros and return would help, but the residual value creeps in when there are non-zero values and then the last term gets the tiny value that propagate to the end of the day (where before the last hours of the day were zero).  

I'll make a second commit with the zero return active to see if that helps speed-wise - penalty to check, but avoid doing the moving avg on a bunch of zeros.